pkg/cnb/image_rebaser_test.go

```diff
@@ -112,3 +80,23 @@ type fakeImageIdentifier struct {
 func (f *fakeImageIdentifier) String() string {
 	return f.identifier
 }
+
+type FakeRemoteImageUtilFactory struct {
+	fakeImages map[string]*fakes.Image
+}
+
+func (f *FakeRemoteImageUtilFactory) newRemote(imageName string, baseImage string, secretRef registry.SecretRef) (imgutil.Image, error) {
+	for imgName, fakeImage := range f.fakeImages {
+		if strings.Contains(baseImage, imgName) {
+			return fakeImage, nil
+		}
+	}
+	return nil, errors.New("not found")
+}
+
+func (f *FakeRemoteImageUtilFactory) NewRemoteReturnsForImageName(imageName string, image *fakes.Image) {
```

Comment:
Does this need to be public?

Reply:
It doesn't need to be exported but, is only accessible inside of the cnb_test package. So, it is not really accessible.